discretisedfield/mesh.py

```diff
@@ -950,6 +950,195 @@ def plane(self, *args, n=None, **kwargs):
+            raise ValueError(
+                "Either one positional argument or a keyword argument can be passed."
+            )
+
+        dim_index = self.region._dim2index(dim)
+
+        if range_ is not None:
+            if isinstance(range_, numbers.Real):
+                # TODO: Some book-keeping in future.
+                selected_value = range_
+            elif isinstance(range_, (tuple, list, np.ndarray)):
+                if len(range_) != 2:
+                    raise ValueError(
+                        "The points along the selected dimension must have two"
+                        " real numbers."
+                    )
+                elif not all(isinstance(point, numbers.Real) for point in range_):
+                    raise TypeError(
+                        f"The elements of {type(range_)} passed as the value of keyword"
+                        " argument must be real numbers."
+                    )
+            else:
+                raise TypeError(
+                    "The value passed to selected dimension must be a tuple, list,"
+                    " array or real number."
+                )
+        else:
+            selected_value = self.region.center[dim_index]
+
+        sub_region = dict()
+        if range_ is None or isinstance(range_, numbers.Real):
+            idxs = [i for i in range(self.region.ndim) if i != dim_index]
+            p_1 = list()
+            p_2 = list()
+            cell = list()
+            dims = list()
+            units = list()
+            for j in idxs:
+                p_1.append(self.region.pmin[j])
+                p_2.append(self.region.pmax[j])
+                cell.append(self.cell[j])
+                dims.append(self.region.dims[j])
+                units.append(self.region.units[j])
+            tol = self.region.tolerance_factor
+
+            if self.subregions is not None:
+                for key, subreg in self.subregions.items():
+                    if (
+                        selected_value >= subreg.pmax[dim_index]
+                        or selected_value < subreg.pmin[dim_index]
+                    ):
+                        continue
+                    else:
+                        sub_p_1 = list()
+                        sub_p_2 = list()
+                        sub_dims = list()
+                        sub_units = list()
+                        for j in idxs:
+                            sub_p_1.append(subreg.pmin[j])
+                            sub_p_2.append(subreg.pmax[j])
+                            sub_dims.append(subreg.dims[j])
+                            sub_units.append(subreg.units[j])
+                        sub_tol = subreg.tolerance_factor
+                        sub_region[key] = df.Region(
+                            p1=sub_p_1,
+                            p2=sub_p_2,
+                            dims=sub_dims,
+                            units=sub_units,
+                            tolerance_factor=sub_tol,
```

Comment:
```suggestion
                        sub_dims = list()
                        sub_units = list()
                        for j in idxs:
                            sub_p_1.append(subreg.pmin[j])
                            sub_p_2.append(subreg.pmax[j])
                            sub_dims.append(subreg.dims[j])
                            sub_units.append(subreg.units[j])
                        sub_tol = subreg.tolerance_factor
                        sub_region[key] = df.Region(
                            p1=sub_p_1,
                            p2=sub_p_2,
                            dims=sub_dims,
                            units=sub_units,
                            tolerance_factor=sub_tol,
```
```suggestion
                        for j in idxs:
                            sub_p_1.append(subreg.pmin[j])
                            sub_p_2.append(subreg.pmax[j])
                        sub_region[key] = df.Region(
                            p1=sub_p_1,
                            p2=sub_p_2,
```

Reply:
We have changed the default behaviour to force subregions to have the same `units`, `dims`, and `tol` as the main region so we don't need to set it.